src/tls.c

```diff
@@ -786,6 +787,19 @@ static int connTLSWrite(connection *conn_, const void *data, size_t data_len) {
     return ret;
 }
 
+static int connTLSWritev(connection *conn_, const struct iovec *iov, int iovcnt) {
+    size_t cum = 0;
+    for (int i = 0; i < iovcnt; i++) cum += iov[i].iov_len;
+
+    char buf[cum];
+    size_t offset = 0;
+    for (int i = 0; i < iovcnt; i++){
+        memcpy(buf + offset, iov[i].iov_base, iov[i].iov_len);
+        offset += iov[i].iov_len;
+    }
```

Comment:
This looks a bit odd, as it recopies all the memory to a new buf.

Reply:
Yep..., this is more like an implementation of compatibility for TLS cuz there is no such a system call of `SSL_writev()`.

Reply:
If this is done, the performance of tls may be affected, we may need more benchmarking to verify it.

Reply:
The current implementation for TLS increases memory copies but reduces system calls, so I think it's still an uncertain impact, we can have some benchmarks later to check it out.

Reply:
We must distinct this feature to small packets, otherwise we risk with  stack-overflow here and multiple redundant copies of big chunks such that each time a small portion will be actually sent. 
If you still not convinced, please set-and-get a key of size 1MB and enter follow the code to get better sense.